oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/logout/OidcClientInitiatedLogoutSuccessHandler.java

```diff
@@ -101,6 +122,26 @@ private String idToken(Authentication authentication) {
 	 */
 	public void setPostLogoutRedirectUri(URI postLogoutRedirectUri) {
 		Assert.notNull(postLogoutRedirectUri, "postLogoutRedirectUri cannot be null");
-		this.postLogoutRedirectUri = postLogoutRedirectUri;
+		this.postLogoutRedirectUriTemplate = postLogoutRedirectUri.toASCIIString();
+	}
+
+	/**
+	 * Set the post logout redirect uri template to use. Supports the {@code "{baseUrl}"}
+	 * placeholder, for example:
+	 *
+	 * <pre>
+	 * 	handler.setPostLogoutRedirectUriTemplate("{baseUrl}");
+	 * </pre>
+	 *
+	 * will make so that {@code post_logout_redirect_uri} will be set to the base url for the client
+	 * application.
+	 *
+	 * @param postLogoutRedirectUriTemplate - A template for creating the {@code post_logout_redirect_uri}
+	 * query parameter
+	 * @since 5.3
+	 */
+	public void setPostLogoutRedirectUriTemplate(String postLogoutRedirectUriTemplate) {
```

Comment:
I don't really like this option of adding another configuration method that is similar to the existing `setPostLogoutRedirectUri(URI postLogoutRedirectUri)`. It may confuse the user.

I thought of an alternative method that we can try out. What if we use a delegation-based strategy where the delegator calculates the `baseUrl` and then calls `setPostLogoutRedirectUri()` on the delegate before delegating. I think this would work and might be enough to provide @mraible as a custom configuration on his end and no changes on our end. Let's try this out in a sample and see how it goes.

Reply:
Since the decision would be based on the incoming request, and this is a setter, I think that would cause some concurrency issues.

> I don't really like this option of adding another configuration method

Based on an offline convo with @jgrandja, I think what we could do is deprecate the older method that takes a `URI` and rename this one.

Other classes that have configurable URI values in Spring Security take a String like `mvcMatchers`, `ClientRegistration`, and `RelyingPartyRegistration` because of this expressed desire to be able to have placeholders.